src/arviz_plots/plots/prior_posterior_plot.py

```diff
@@ -150,8 +150,8 @@ def plot_prior_posterior(
         .assign_coords(sample=("sample", np.arange(num_samples)))
     )
 
-    distribution = concat([ds_prior, ds_posterior], dim="Groups").assign_coords(
-        {"Groups": ["prior", "posterior"]}
+    distribution = concat([ds_prior, ds_posterior], dim="Group").assign_coords(
+        {"Group": ["prior", "posterior"]}
```

Comment:
We always use all lowercase for dimension names. How would you feel about using lowercase here and then having the legend generating function use `.capitalize` so the legend still looks the same? Maybe we could even replace underscores by spaces?

I know any string is a valid dimension name, so we could use uppercase and even spaces in them, but it is a bit annoying for some reason.

Note: we don't need to do this in this PR. The legend generating function already uses the labeller to generate the labels of the legend elements, we could add a new method to the labellers to take care of legend title.

Second note: it looks like the change `Groups` -> `Group` is the reason a couple tests are now failing.